scripts/installers/make-exe

```diff
@@ -77,6 +80,16 @@ def copy_file(src, dst):
     shutil.copy2(src, dst)
 
 
+def update_metadata(dirname):
+    print('Update metadata distribution source to %s' % DISTRIBUTION_SOURCE)
+    metadata_file = os.path.join(dirname, 'awscli', 'data', 'metadata.json')
+    with open(metadata_file) as f:
+        metadata = json.load(f)
+    metadata['distribution_source'] = DISTRIBUTION_SOURCE
+    with open(metadata_file, 'w') as f:
+        json.dump(metadata, f)
+
+
```

Comment:
So for this change, I'm thinking we do the following:

1. Move and generalize this `update_metadata` logic  to `awscli/scripts/utils.py` and import it from there.
2. Remove the `sed` logic in the `Dockerfile` and use this shared utility in the `make-docker` by unzipping the exe, updating the metadata file in the exe, and then zipping it back up when used in the Docker build context.

I think this makes it a bit cleaner as it will allow us to isolate this metadata update logic in one place for the various installers.